aya/src/maps/mod.rs

```diff
@@ -185,11 +188,17 @@ pub enum MapError {
 }
 
 /// A map file descriptor.
-pub struct MapFd(RawFd);
+pub struct MapFd<'f>(BorrowedFd<'f>);
```

Comment:
this is a breaking change in that prior to this PR `MapFd` would not have a lifetime associated to the map it came from whereas now it does. Having a lifetime associated with it is more "correct" though in that before a map could be dropped (closing the file descriptor) but a `MapFd` could still hang around. We could make `MapFd` have a cloned file descriptor or use `Arc<OwnedFd>` in maps but neither of those felt quite right so I left it at this "dumb" way for now. Feedback here on what feels more appropriate for maintainers would be appreciated.